### ISSUE TITLE
[Discussion] - XCB instead of Xlib?

### ISSUE THREAD

**Issue by dalmoz** [mackstann/tinywm#2]
Will it be more efficient to make the project dependent on _XCB_ instead of _Xlib_?

In terms of -
1) Fewer lines of code
2) Resource-consumption-wise

**Comment by mackstann** [comment]
My vague impression of XCB is that it would probably require more lines of code. I could definitely be wrong though.

Resource consumption -- what resource consumption? :-)

I just googled it and it looks like someone has beaten us to the punch: https://github.com/rtyler/tinywm-ada/blob/master/tinywm-xcb.c

**Comment by ar1ja** [comment]
> My vague impression of XCB is that it would probably require more lines of code. I could definitely be wrong though.
> 
> Resource consumption -- what resource consumption? :-)
> 
> I just googled it and it looks like someone has beaten us to the punch: https://github.com/rtyler/tinywm-ada/blob/master/tinywm-xcb.c

> Resource consumption -- what resource consumption? :-)

Yeah I agree (so small it's barely noticable), but as XCB is smaller the binary would not be as big (probably) and it might be even lighter on ram :+1:
